src/profile-logic/marker-data.js

```diff
@@ -459,6 +461,30 @@ export function deriveMarkersFromRawMarkerTable(
         break;
       }
 
+      case 'IPC': {
+        const times = ipcCorrelations[[threadId, i]];
```

Comment:
I'm a bit confused by the `ipcCorrelations` structure, and how it's working with taking a tuple like `[threadId, i]`. Also, it's best to avoid small allocations if possible in hot code, which all this marker iteration is going to be hot.

Reply:
Oh wow, I guess this runs `toString()` on the array, and uses that as a key. I'd prefer a more straightforward approach with string concatenation. This code will be hot enough to justify writing C-like code as much as possible.

Reply:
Yeah, this was originally going to be a `Map` with a pair as the key, but I forgot that comparison of two JS arrays compares their memory locations, not their values. So for the time being, I figured the easiest change was to use an `Object` instead and worry about doing it right later. :)